GitUI/CommandsDialogs/RevisionFileTreeControl.cs

```diff
@@ -550,51 +550,51 @@ private void FileTreeContextMenu_Opening(object sender, System.ComponentModel.Ca
             var isFile = itemSelected && gitItem.ObjectType == GitObjectType.Blob;
             var isFolder = itemSelected && gitItem.ObjectType == GitObjectType.Tree;
             var isFileOrFolder = isFile || isFolder;
+
+            // Many items does not make sense if a local file does not exist, why this is used for Enabled
             var isExistingFileOrDirectory = itemSelected && FormBrowseUtil.IsFileOrDirectory(_fullPathResolver.Resolve(gitItem.FileName));
 
-            if (itemSelected && gitItem.ObjectType == GitObjectType.Commit)
-            {
-                openSubmoduleMenuItem.Visible = true;
-                if (!openSubmoduleMenuItem.Font.Bold)
-                {
-                    openSubmoduleMenuItem.Font = new Font(openSubmoduleMenuItem.Font, FontStyle.Bold);
-                }
-            }
-            else
+            openSubmoduleMenuItem.Visible = itemSelected && gitItem.ObjectType == GitObjectType.Commit && isExistingFileOrDirectory;
+            if (openSubmoduleMenuItem.Visible && !openSubmoduleMenuItem.Font.Bold)
             {
-                openSubmoduleMenuItem.Visible = false;
+                openSubmoduleMenuItem.Font = new Font(openSubmoduleMenuItem.Font, FontStyle.Bold);
             }
 
-            saveAsToolStripMenuItem.Visible = isFile;
-            resetToThisRevisionToolStripMenuItem.Visible = isFileOrFolder && !Module.IsBareRepository();
-            toolStripSeparatorFileSystemActions.Visible = isFileOrFolder;
-
-            copyFilenameToClipboardToolStripMenuItem.Visible = itemSelected;
-            fileTreeOpenContainingFolderToolStripMenuItem.Enabled = isExistingFileOrDirectory;
-            fileTreeArchiveToolStripMenuItem.Enabled = itemSelected;
-            fileTreeCleanWorkingTreeToolStripMenuItem.Visible = isFileOrFolder;
-            fileTreeCleanWorkingTreeToolStripMenuItem.Enabled = isExistingFileOrDirectory;
-
-            fileHistoryToolStripMenuItem.Enabled = itemSelected;
-            blameToolStripMenuItem1.Visible = isFile;
+            // Diff with workTree (some tools like kdiff3 and meld allows diff to NUL)
+            openWithDifftoolToolStripMenuItem.Visible = isFile;
+            resetToThisRevisionToolStripMenuItem.Visible = itemSelected && !Module.IsBareRepository();
+            toolStripSeparatorTopActions.Visible = itemSelected && ((gitItem.ObjectType == GitObjectType.Commit && isExistingFileOrDirectory)
+                                                                    || !Module.IsBareRepository()
+                                                                    || isFile);
```

Comment:
This condition is too complex IMO.
```suggestion
            toolStripSeparatorTopActions.Visible = openSubmoduleMenuItem.Visible
                || openWithDifftoolToolStripMenuItem.Visible
                || resetToThisRevisionToolStripMenuItem.Visible;
```

Reply:
That is what I would like too, but Visible is always false when reading.
(It would be possible to set Enabled or a separate variable of course).